tests/apps/core/test_helpers.py

```diff
@@ -219,3 +225,120 @@ def test_helpers_create_i18n_page_published(self, mock_publish):
+        """The site argument is required for recursive page creation."""
+        with self.assertRaises(TypeError) as context:
+            # pylint: disable=no-value-for-parameter
+            recursive_page_creation(pages_info=PAGES_INFO)
+        self.assertEqual(
+            str(context.exception),
+            "recursive_page_creation() missing 1 required positional argument: 'site'",
+        )
+
+    def test_helpers_recursive_page_creation_missing_pages_info_argument(self):
+        """The pages_info argument is required for recursive page creation."""
+        site = Site.objects.get(id=1)
+        with self.assertRaises(TypeError) as context:
+            # pylint: disable=no-value-for-parameter
+            recursive_page_creation(site=site)
+        self.assertEqual(
+            str(context.exception),
+            "recursive_page_creation() missing 1 required positional argument: 'pages_info'",
+        )
+
+    def test_helpers_recursive_page_creation_with_defaults(self):
+        """Create defaults pages."""
+        site = Site.objects.get(id=1)
+        self.assertEqual(Page.objects.count(), 0)
+        recursive_page_creation(site=site, pages_info=PAGES_INFO)
+        self.assertEqual(Page.objects.filter(reverse_id="home").count(), 2)
+        self.assertEqual(Page.objects.filter(reverse_id="blogposts").count(), 2)
+        self.assertEqual(Page.objects.filter(reverse_id="categories").count(), 2)
+        self.assertEqual(Page.objects.filter(reverse_id="courses").count(), 2)
+        self.assertEqual(Page.objects.filter(reverse_id="organizations").count(), 2)
+        self.assertEqual(Page.objects.filter(reverse_id="persons").count(), 2)
+        self.assertEqual(Page.objects.count(), 12)
+
+    def test_helpers_recursive_page_creation_can_be_run_multiple_times(self):
+        """Ensure we can run the recursive_page_creation helper multiple times with the same
+        parameters without failing or creating new pages."""
+        site = Site.objects.get(id=1)
+        self.assertEqual(Page.objects.count(), 0)
+        recursive_page_creation(site=site, pages_info=PAGES_INFO)
+        recursive_page_creation(site=site, pages_info=PAGES_INFO)
+        self.assertEqual(Page.objects.filter(reverse_id="home").count(), 2)
+        self.assertEqual(Page.objects.filter(reverse_id="blogposts").count(), 2)
+        self.assertEqual(Page.objects.filter(reverse_id="categories").count(), 2)
+        self.assertEqual(Page.objects.filter(reverse_id="courses").count(), 2)
+        self.assertEqual(Page.objects.filter(reverse_id="organizations").count(), 2)
+        self.assertEqual(Page.objects.filter(reverse_id="persons").count(), 2)
+        self.assertEqual(Page.objects.count(), 12)
+
+    def test_helpers_recursive_page_creation_recursiveness(self):
+        """Test embedded pages creation."""
+        site = Site.objects.get(id=1)
+
+        pages_info = PAGES_INFO.copy()
+        pages_info.update(
```

Comment:
This is quite a slow test. So I think you should test only one embedded page here. The above test for the full PAGES_INFO is also testing on too many pages.